src/common/procedure/src/store.rs

```diff
@@ -119,9 +127,50 @@ impl ProcedureStore {
         Ok(())
     }
 
-    /// Load uncommitted procedures from the storage.
-    pub(crate) async fn load_messages(&self) -> Result<HashMap<ProcedureId, ProcedureMessage>> {
-        let mut messages = HashMap::new();
+    /// Delete states of procedure from the storage.
+    pub(crate) async fn delete_procedure(&self, procedure_id: ProcedureId) -> Result<()> {
+        let path = proc_path!("{procedure_id}/");
+        // TODO(yingwen): We can optimize this to avoid reading the value.
+        let mut key_values = self.0.walk_top_down(&path).await?;
+        // 8 should be enough for most procedures.
+        let mut step_keys = Vec::with_capacity(8);
+        let mut finish_keys = Vec::new();
+        while let Some((key, _)) = key_values.try_next().await? {
+            let Some(curr_key) = ParsedKey::parse_str(&key) else {
+                logging::warn!("Unknown key while deleting procedures, key: {}", key);
```

Comment:
adds the procedure id to log.

Reply:
The key already contains the procedure id.